motionplan/motionPlanner.go

```diff
@@ -569,9 +573,10 @@ func CheckPlan(
 
 			modifiedSegment := &ik.State{Frame: sf, Position: poseInPath}
 			if isValid, _ := sfPlanner.planOpts.CheckStateConstraints(modifiedSegment); !isValid {
-				return fmt.Errorf("found collsion between positions %v and %v", currentPose.Point(), nextPose.Point())
+				return poseInPath, fmt.Errorf("found collision between positions %v and %v", currentPose.Point(), nextPose.Point())
```

Comment:
[note] returning pose here to allow the position to be compared to the original position of the robot and act accordingly (i.e. for explore, if an obstacle is detected very far away, still move towards it until it is within range to be an issue)

Reply:
I would add something like this as a comment above the function.  Also want to call out that specifically this is the first interpolated point to be found in collision and does not represent an accurate distance of the offending obstacle.

Reply:
Comment added